builtin/providers/aws/resource_aws_organization_account.go

```diff
@@ -0,0 +1,230 @@
+		Target:       []string{"SUCCEEDED"},
+		Refresh:      resourceAwsOrganizationAccountStateRefreshFunc(conn, requestId),
+		PollInterval: 10 * time.Second,
+		Timeout:      5 * time.Minute,
+	}
+	stateResp, stateErr := stateConf.WaitForState()
+	if stateErr != nil {
+		return fmt.Errorf(
+			"Error waiting for account request (%s) to become available: %s",
+			requestId, stateErr)
+	}
+
+	// Store the ID
+	accountId := stateResp.(*organizations.CreateAccountStatus).AccountId
+	d.SetId(*accountId)
+
+	log.Printf("[INFO] Account ID: %s", d.Id())
+
+	return resourceAwsOrganizationAccountUpdate(d, meta)
+}
+
+func resourceAwsOrganizationAccountRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).orgsconn
+	describeOpts := &organizations.DescribeAccountInput{
+		AccountId: aws.String(d.Id()),
+	}
+	resp, err := conn.DescribeAccount(describeOpts)
+	if err != nil {
+		if orgerr, ok := err.(awserr.Error); ok && orgerr.Code() == "AccountNotFoundException" {
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	d.Set("arn", resp.Account.Arn)
+	d.Set("joined_method", resp.Account.JoinedMethod)
+	d.Set("joined_timestamp", resp.Account.JoinedTimestamp)
+	d.Set("name", resp.Account.Name)
+	d.Set("status", resp.Account.Status)
+	return nil
+}
+
+func resourceAwsOrganizationAccountUpdate(d *schema.ResourceData, meta interface{}) error {
+	return resourceAwsOrganizationAccountRead(d, meta)
+}
+
+func resourceAwsOrganizationAccountDelete(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[WARN] Organization member accounts must be deleted from the web console.")
+	d.SetId("")
+	return nil
+}
+
+// resourceAwsOrganizationAccountStateRefreshFunc returns a resource.StateRefreshFunc
+// that is used to watch a CreateAccount request
+func resourceAwsOrganizationAccountStateRefreshFunc(conn *organizations.Organizations, id string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		opts := &organizations.DescribeCreateAccountStatusInput{
+			CreateAccountRequestId: aws.String(id),
+		}
+		resp, err := conn.DescribeCreateAccountStatus(opts)
+		if err != nil {
+			if orgerr, ok := err.(awserr.Error); ok && orgerr.Code() == "CreateAccountStatusNotFoundException" {
+				resp = nil
+			} else {
+				log.Printf("Error on OrganizationAccountStateRefresh: %s", err)
+				return nil, "", err
+			}
+		}
+
+		if resp == nil {
+			// Sometimes AWS just has consistency issues and doesn't see
+			// our instance yet. Return an empty state.
+			return nil, "", nil
+		}
+
+		accountStatus := resp.CreateAccountStatus
+		if *accountStatus.State == "FAILED" {
+			return nil, *accountStatus.State, fmt.Errorf(*accountStatus.FailureReason)
+		}
+		return accountStatus, *accountStatus.State, nil
+	}
+}
+
+func validateAwsOrganizationAccountEmail(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	if !regexp.MustCompile(`^[^\s@]+@[^\s@]+\.[^\s@]+$`).MatchString(value) {
```

Comment:
I would recommend [`net/mail.ParseAddress`](https://golang.org/pkg/net/mail/#ParseAddress) here. What do you think?